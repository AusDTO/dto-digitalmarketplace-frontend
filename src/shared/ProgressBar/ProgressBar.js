import React from 'react';
import classNames from 'classnames';

const ProgressBar = ({ value, max }) => {
  const className = classNames('progress-bar__progress', {
    'progress-bar--max': max <= value
  });
  
  let progress = value / max * 100;
  if (progress > 100) {
    progress = 100;
  }

  return (
    <div className="progress-bar">
      <div className={className} style={{ width: `${progress}%`}}></div>
    </div>
  )
};

ProgressBar.propTypes = {
  value: React.PropTypes.number.isRequired,
  max: React.PropTypes.number.isRequired
};

export default ProgressBar;