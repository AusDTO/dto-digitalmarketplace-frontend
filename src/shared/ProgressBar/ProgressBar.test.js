import React from 'react';
import renderer from 'react-test-renderer';

import ProgressBar from './ProgressBar';

test('Maxed out progressbar', () => {
  const component = renderer.create(
    <ProgressBar value={5} max={5} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Incomplete progress', () => {
  const component = renderer.create(
    <ProgressBar value={3} max={5} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('value greater than max', () => {
  const component = renderer.create(
    <ProgressBar value={8} max={5} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});