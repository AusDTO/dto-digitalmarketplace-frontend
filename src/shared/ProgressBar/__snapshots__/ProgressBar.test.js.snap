exports[`test Incomplete progress 1`] = `
<div
  className="progress-bar">
  <div
    className="progress-bar__progress"
    style={
      Object {
        "width": "60%",
      }
    } />
</div>
`;

exports[`test Maxed out progressbar 1`] = `
<div
  className="progress-bar">
  <div
    className="progress-bar__progress progress-bar--max"
    style={
      Object {
        "width": "100%",
      }
    } />
</div>
`;

exports[`test value greater than max 1`] = `
<div
  className="progress-bar">
  <div
    className="progress-bar__progress progress-bar--max"
    style={
      Object {
        "width": "100%",
      }
    } />
</div>
`;
